apps/dashboard/app/models/files.rb

```diff
@@ -21,6 +21,10 @@ def self.files_to_zip(path)
   def ls(dirpath)
     Pathname.new(dirpath).each_child.map do |path|
       Files.stat(path)
+    end.select do |stats|
+      valid_encoding = stats[:name].to_s.valid_encoding?
+      Rails.logger.warn("Not showing file '#{stats[:name]}' because it is not a UTF-8 filename.") unless valid_encoding
+      valid_encoding
```

Comment:
This looks good, but for readability could we just do:
```ruby
valid_encoding = stats[:name].to_s.valid_encoding?

unless valid_encoding
   Rails.logger.warn("Not showing file '#{stats[:name]}' because it is not a UTF-8 filename.")
end

valid_encoding
```